### PR TITLE
Cadsobs adaptor

### PR DESCRIPTION
**Pull request by garciampred**

### CODE REVIEW

File: cads_adaptors/adaptors/cadsobs.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class ObservationsAdaptor(AbstractCdsAdaptor):
+    def retrieve(self, request):
+        from cdsobs.retrieve.api import retrieve_observations
+        from cdsobs.retrieve.models import RetrieveArgs
+        from cdsobs.utils.utils import get_database_session
+
+        # Maps observation_type to source
+        mapped_request = apply_mapping(request, self.mapping)
```
Comment (EddyCMWF) [inline cads_adaptors/adaptors/cadsobs.py:14]: Could you merge from main and then change this to:
```suggestion
        self._pre_retrieve(request=request)
```

**Review by EddyCMWF — CHANGES_REQUESTED** [review]
When the merge from main is done and the small change I requested is made I am happy to merge

**Comment by EddyCMWF** [comment]
Actually, there are quite a few references to mapped_request, can you update them to reference the class attribute, self.mapped_request

**Comment by garciampred** [comment]
Like this? I just assigned mapped_request = self.mapped_request, which is what I usually do to avoid repeating self everywhere.

**Review by EddyCMWF — APPROVED** [review]

**Comment by EddyCMWF** [comment]
Thanks @garciampred !

**Pull request merged**